ibis/expr/types/relations.py

```diff
@@ -70,7 +69,9 @@ def f(  # noqa: D417
         | Sequence[
             str | tuple[str | ir.Column, str | ir.Column] | ir.BooleanValue
         ] = (),
-        suffixes: tuple[str, str] = ("_x", "_y"),
+        *,
+        lname: str = "",
+        rname: str = "{name}_right",
```

Comment:
I like it, it certainly provides more flexibility.

Reply:
> To get the exact same behavior as before, you can pass in lname="{name}_x", rname="{name}_y" to join calls.

Could we unpack a `suffixes` keyword argument if it is passed to the method?

```py
if 'suffixes' in kwargs:
    # raise deprecation warning
    lsuffix, rsuffix = suffixes
    lname, rname = "{name}" + lsuffix, "{name}" + rsuffix
```

If the workaround is that simple then I'd favor having a deprecation notice for one major release.

Reply:
Since the default behavior has changed, deprecating this way would work for users that were explicitly passing in `suffixes`, but would still be a breaking change for the majority of users that were relying on the default behavior.

Since we can't deprecate the change in default behavior, I'd rather this be a breaking change for all cases.